### PR TITLE
Initial post for ESP8266 and MicroPython

### PR DESCRIPTION
**Pull request by fabaff**
This is work-in-progress.

### CODE REVIEW

**Comment by balloob** [comment]
We can use

import urequests 
urequests.get(...)

Is the firmware already generally available?

**Comment by fabaff** [comment]
> Is the firmware already generally available? 

I think only the participant in the kickstarter campaign have access to the firmware at the moment. Would be best if the post don't get published till the firmware is available for everyone.

**Comment by balloob** [comment]
Agree

On Fri, Jun 3, 2016 at 6:43 AM, Fabian Affolter <notifications@github.com>
wrote:

> Is the firmware already generally available?
>
> I think only the participant in the kickstarter campaign have access to
> the firmware at the moment. Would be best if the post don't get published
> till the firmware is available for everyone.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/home-assistant.io/pull/532#issuecomment-223582107>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABYJ2hHuvwiGthMvxFzX0oT7vPEP635rks5qIC92gaJpZM4ItaXP>
> .
>



-- 
PaulusSchoutsen.nl
It's nice to be important but it's more important to be nice.

**Comment by balloob** [comment]
Latest build has DHT11/DHT22 sensor support. So we can do a full copy of the old article

**Comment by fabaff** [comment]
Got the e-mail about v1.8.1 a couple of hours ago.

**Comment by balloob** [comment]
What's with all the commits? This branch needs some clean up.

File: source/_posts/2016-07-21-esp8266-and-micropython-part1.markdown [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+author: Fabian Affolter
+comments: true
+categories: How-To
+og_image: /images/blog/2016-07-micropython/social.png
+---
+
+<img src='/images/blog/2016-07-micropython/micropython.png' style='clear: right; border:none; box-shadow: none; float: right; margin-bottom: 12px;' width='200' />
+The first release of Micropython for ESP8266 was delivered a couple of weeks ago. The [documentation](http://docs.micropython.org/en/latest/esp8266/esp8266_contents.html) covers a lot of ground. This post is providing only a little summary which should get you started.
+
+If you don't participated in the KickStarter campaign only the source code was available. This meant that you needed to build the firmware on your own. As of now the [pre-built firmware](https://micropython.org/download/#esp8266) is available for the public.
```
Comment (balloob) [inline source/_posts/2016-07-21-esp8266-and-micropython-part1.markdown]: Until a couple of weeks ago, the pre-built MicroPython binary for the ESP8266 was only available to backers. This has changed now and it is available to the public for [download](https://micropython.org/download/#esp8266).

File: source/_posts/2016-07-21-esp8266-and-micropython-part1.markdown [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+esptool.py v1.2-dev
+Connecting...
+Running Cesanta flasher stub...
+Flash params set to 0x0020
+Writing 540672 @ 0x0... 540672 (100 %)
+Wrote 540672 bytes at 0x0 in 13.1 seconds (330.8 kbit/s)...
+Leaving...
+```
+
+Now reset the device. You should then be able to use a terminal program like `minicom` or `picocom` to connect and get the [REPL (Read Evaluate Print Loop)](http://docs.micropython.org/en/latest/esp8266/esp8266/tutorial/repl.html#getting-a-micropython-repl-prompt) prompt.
```
Comment (balloob) [inline source/_posts/2016-07-21-esp8266-and-micropython-part1.markdown]: On a mac: `screen /dev/tty.SLAB_USBtoUART 115200`. On Windows you can use Putty to open a serial connection. Best to link to a MicroPython page with per OS instructions ?

**Comment by fabaff** [comment]
Ups, that merge went wrong

File: source/_posts/2016-07-21-esp8266-and-micropython-part1.markdown (hunk 2 above)
Comment (fabaff) [inline source/_posts/2016-07-21-esp8266-and-micropython-part1.markdown]: [REPL over the serial port](http://docs.micropython.org/en/latest/esp8266/esp8266/tutorial/repl.html#repl-over-the-serial-port) is pretty short. At the moment I can't find OS specific instructions in the Micropython docs. I guess the reason for this is that WebREPL is sort of default/recommended way.

File: source/_posts/2016-07-21-esp8266-and-micropython-part1.markdown (hunk 1 above)
Comment (fabaff) [inline source/_posts/2016-07-21-esp8266-and-micropython-part1.markdown]: Changed

**Pull request merged**